include/mav/Network.h

```diff
@@ -170,7 +172,12 @@ namespace mav {
         std::function<void(const std::shared_ptr<Connection>&)> _on_connection_lost;
 
         void _sendMessage(Message &message, const ConnectionPartner &partner) {
-            int wire_length = static_cast<int>(message.finalize(_seq++, _own_id));
+            const bool sign = bool(_get_timestamp_function);
+            int wire_length = static_cast<int>(message.finalize(_seq++, _own_id, sign));
+            if (sign) {
+                message.sign(_key, _get_timestamp_function());
+                wire_length += MessageDefinition::SIGNATURE_SIZE;
+            }
```

Comment:
Why can't `finalize` do this?

Reply:
Because `finalize()` doesn't have access to `_key` or `_get_timestamp_function()`. Those exist at the network level.

Reply:
Addressed. See other thread about sign() vs finalize().